### ISSUE TITLE
expandRanges is incompatible with prerelease-tags

### ISSUE THREAD

**Issue by heikomat** [snyamathi/semver-intersect#4]
when asking semver-intersect if these intersect:
```
intersect('^1.0.0-rc3', '^1.0.0-rc4')
```

i get the following error:
```
Error: Range >=1.0.0-rc4 is not compatible with <2.0.0
    at /Users/heiko/Desktop/Projekte/5minds/minstall/node_modules/semver-intersect/semver-intersect.js:71:19
```

The reason for that is, that expandRanges splits the result of `semver.validRange(range)` here: https://github.com/snyamathi/semver-intersect/blob/818619388892870af2c3a14ef4a9ee0632d22e0f/src/semver-intersect.js#L61

In this example, `semver.vaildRange('^1.0.0-rc3')` results in `>=1.0.0-rc3 <2.0.0`, which `1.0.0-rc4` does satisfy, but `expandRanges` splits them into two seperate ranges:
- `>=1.0.0-rc3` and
- `<2.0.0`

However, `1.0.0-rc4` only satisfies the first range, not the second one! This is explained in the README of semver, [here](https://github.com/npm/node-semver#prerelease-tags):

> If a version has a prerelease tag (for example, 1.2.3-alpha.3) then it will only be allowed to satisfy comparator sets if at least one comparator with the same [major, minor, patch] tuple also has a prerelease tag.

**Comment by snyamathi** [comment]
Thanks @heikomat - this is indeed a bug that should be fixed.  If you're comfortable poking around and would like to submit a PR then please let me know.  Otherwise I'll try to get to this in the next day or two.

**Comment by snyamathi** [comment]
https://github.com/snyamathi/semver-intersect/pull/5

**Comment by snyamathi** [comment]
Fixed in version 1.3.0

**Comment by heikomat** [comment]
@snyamathi This fixed the problem i had in my package. Thanks a ton for the fast reaction and smooth fix, appreciate it :)
